pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -359,3 +377,41 @@ func TestEventNotCreated(t *testing.T) {
 	}
 	tc.runTest(t)
 }
+
+func TestCPUUtilization(t *testing.T) {
+	tc := testCase{
+		minReplicas:     1,
+		maxReplicas:     5,
+		initialReplicas: 1,
+		desiredReplicas: 2,
+		CPUTarget:       0.5,
+		CPURequest:      resource.MustParse("0.2"),
+		reportedLevels:  []uint64{200},
+	}
+	tc.runTest(t)
+}
+
+func TestMissingCPURequest(t *testing.T) {
+	tc := testCase{
+		minReplicas:     1,
+		maxReplicas:     5,
+		initialReplicas: 1,
+		desiredReplicas: 1,
+		CPUTarget:       0.5,
+		reportedLevels:  []uint64{200},
+	}
+	tc.runTest(t)
+}
+
+func TestCPUUtilizationTolerance(t *testing.T) {
+	tc := testCase{
+		minReplicas:     1,
+		maxReplicas:     5,
+		initialReplicas: 1,
+		desiredReplicas: 1,
+		CPUTarget:       0.5,
+		CPURequest:      resource.MustParse("0.2"),
+		reportedLevels:  []uint64{105, 109, 107},
```

Comment:
I think this test is not checking tolerance. The number of entries in reportedLevels should match initialReplicas.